packages/s3-credentials-endpoint/index.js

```diff
@@ -170,30 +150,36 @@ async function ensureAuthorizedOrRedirect(req, res, next) {
     req.authorizedMetadata = { userName: 'unauthenticated user' };
     return next();
   }
-  req.earthdataLoginClient = buildEarthdataLoginClient();
+
+  const {
+    accessTokenModel,
+    oauthClient,
+  } = await getConfigurations();
+
+  req.oauthClient = oauthClient;
 
   if (isTokenAuthRequest(req)) {
     return handleTokenAuthRequest(req, res, next);
   }
-  const {
-    accessTokenModel,
-    authClient,
-  } = getConfigurations();
-  const redirectURLForAuthorizationCode = authClient.getAuthorizationUrl(req.path);
+
+  if (isAuthBearTokenRequest(req)) {
+    return handleAuthBearerToken(req, res, next);
+  }
+
+  const redirectURLForAuthorizationCode = oauthClient.getAuthorizationUrl(req.path);
   const accessToken = req.cookies.accessToken;
   if (!accessToken) return res.redirect(307, redirectURLForAuthorizationCode);
 
   let accessTokenRecord;
   try {
     accessTokenRecord = await accessTokenModel.get({ accessToken });
   } catch (error) {
-    if (error instanceof RecordDoesNotExist) {
-      return res.redirect(307, redirectURLForAuthorizationCode);
+    if (!(error instanceof RecordDoesNotExist)) {
+      throw error;
     }
-    throw error;
   }
 
-  if (isAccessTokenExpired(accessTokenRecord)) {
+  if (!accessToken || !accessTokenRecord || isAccessTokenExpired(accessTokenRecord)) {
```

Comment:
`if (!accessToken ||...)`
is handled on line above https://github.com/nasa/cumulus/pull/3078/files#diff-3bb3b5cbe8f7561528077af0e7c41cac9db4d4b8728f061d53dfec432fab8c61R171 right?

Reply:
good catch!  Updated